chore: migrate stacks-ui part 4 + 5

pete-watters opened the pull request:
> Try out this version of Leather — [download extension builds](https://github.com/leather-wallet/extension/actions/runs/6828631527).<!-- Sticky Header Marker -->

fbwoolf commented:
@pete-watters I think I am done addressing comments on my branch. You'll have to rebase here again, sorry. Hoping that branch is done.

fbwoolf commented on src/app/components/bitcoin-fees-list/components/fees-list-item.tsx:
```diff
@@ -19,22 +19,24 @@ export function FeesListItem({
-      px="base"
-      py="extra-loose"
+      px="space.04"
+      py="space.06"
```
Thx, I think I missed these. 👍

pete-watters replied:
No worries. There's so many. I wonder if it would be more efficient to leave them alone, then do a big `find+replace` at the end when we have moved all the components. 

I guess no harm to change ones as we go too

fbwoolf commented on src/app/components/crypto-assets/components/asset-caption.tsx:
```diff
@@ -1,7 +1,9 @@
-import { styled } from 'leather-styles/jsx';
+// #4476 TODO: test this tooltip + check with Fara
+// maybe we can replace Tippy with radix tooltip eventually
+import { Tooltip } from '@app/components/tooltip';
```
Yeah, definitely seems like a good idea. 👍

pete-watters replied:
https://github.com/leather-wallet/extension/issues/4506

fbwoolf commented on src/app/components/crypto-assets/components/asset-caption.tsx:
```diff
@@ -17,21 +19,15 @@ export function AssetCaption({ caption, isUnanchored }: AssetCaptionProps) {
-                  as={FiInfo}
-                  ml={1}
-                />
+                <InfoIcon size="12px" ml={1} color="accent.text-subdued" />
```
Think `ml` needs a space token here?

pete-watters replied:
Thanks, I update it. I'll go through my other icons

fbwoolf commented on src/app/components/crypto-assets/crypto-currency-asset/asset-copy-icon.tsx:
```diff
@@ -1,21 +1,19 @@
 }
 
 export function AssetItemCopyIcon({ hasCopied }: AssetItemCopyIconProps) {
+  // #4476 TODO - test and refactor this, so many elements for just an icon!
```
🤣

fbwoolf commented on src/app/pages/bitcoin-contract-list/components/bitcoin-contract-list-item-layout.tsx:
```diff
@@ -39,7 +38,7 @@ export function BitcoinContractListItemLayout({
   return (
     <Flex
-      as={'button'}
+      //  #4476 TODO check with Fara - don't think as button is needed
```
You should be able to turn this into a `styled.button` ...that is typically how I handled the `as="button"` situation.

fbwoolf commented on src/app/pages/choose-account/components/accounts.tsx:
```diff
@@ -21,25 +20,34 @@ import { usePressable } from '@app/components/item-hover';
+// I re-wrote this as it was way over complicated but not sure if its needed
+// - changes color based on whether its loading or not
+// const loadingProps = { color: '#A1A7B3' };
+// const getLoadingProps = (loading: boolean) => (loading ? loadingProps : {});
```
Not sure here either, it must be really old code. It is likely outdated. GitLens says the code was added 4 years ago, haha. It does look like it just change the color during loading.

pete-watters replied:
I tested it, changing the colour and I can't see any difference when loading or not. 

I'm going to delete it, on my head be it if it's an important thing, I'll just re-add it if we need to

fbwoolf commented on src/app/pages/choose-account/components/accounts.tsx:
```diff
@@ -108,10 +113,14 @@ const AddAccountAction = memo(() => {
-        <Text color="currentColor">Generate new account</Text>
-      </Stack>
+      <HStack alignItems="center">
+        <FiPlusCircle size="16px" />
```
Maybe we can just use the `PlusIcon` here?

pete-watters replied:
Great, thanks! I've been copying your icon updates and had meant to ask you this one. 

I'm not even sure where this button is, do you know? It says `Generate new account` but the button I know of is `Create account`

fbwoolf replied:
I believe this is in the transaction request popup when signing into an app.

pete-watters replied:
OK thanks. I'll test there

fbwoolf commented on src/app/pages/receive/components/address-qr-code.tsx:
```diff
@@ -1,16 +1,17 @@
-        foregroundColor: color('invert'),
+        // #4476 TODO check this and ask Fara
+        backgroundColor: 'accent.text-primary',
+        foregroundColor: 'invert',
```
This does look right.

pete-watters replied:
Awesome, thanks! Aysnc work FTW

fbwoolf commented:
Looks like great progress @pete-watters 👍

pete-watters commented:
> ~@pete-watters I think I am done addressing comments on my branch. You'll have to rebase here again, sorry. Hoping that branch is done.~
> 
> Nvm, I rebased and dropped the commits from my branch for you.

Awh thanks! It's no problem to rebase if needed, thats the nature of a task like this

pete-watters commented on src/app/ui/utils/truncate-middle.ts:
```diff
@@ -0,0 +1,23 @@
+function truncateHex(hex: string, offset = 5): string {
```
I just moved this here so we could have the docs in `README` also. I can move it back if we need to

fbwoolf replied:
Seems smart to me to move ui utilities here. 👍

fbwoolf commented:
Noting here, I had a merge conflict I just realized on my new branch I didn't resolve correctly. I will push up a fix to the base branch. It was with creating the `error.label` color. Will fix!

pete-watters commented:
@fbwoolf I've just been pushing stuff here, my outstanding items are listed here: https://github.com/leather-wallet/extension/issues/4381 

I need to stop now as my eyes hurt but we made good progress!

fbwoolf commented:
Merging this into part 3 so I pull the changes into my part 1 branch.
